### PR TITLE
FT: Update Role value check

### PR DESCRIPTION
**Pull request by bennettbuchanan**

### CODE REVIEW

File: lib/api/apiUtils/bucket/models/ReplicationConfiguration.js [hunk 1]
```diff
@@ -122,12 +122,8 @@ class ReplicationConfiguration {
         if (!Role) {
             return errors.MalformedXML;
         }
-        // TODO: Update to validate role priveleges after implemented in Vault.
-        // Role should be an IAM user name.
-        const arr = Role[0].split(':');
-        const isValidARN = arr.length === 7 ||
-            (arr.length === 6 && arr[5].split('/').length === 2);
-        if (!isValidARN) {
+        const regexp = /^arn:aws:iam::.*:role.*/;
```
Comment (jonathan-gramain) [inline lib/api/apiUtils/bucket/models/ReplicationConfiguration.js]: detail: I don't know exactly what is a valid ARN, but you may want to replace `.*` before `:role` with `[^:]*` to exclude the colon character, or even only allow the subset of characters allowed in the ARN path item?

Comment (bennettbuchanan) [inline lib/api/apiUtils/bucket/models/ReplicationConfiguration.js]: Thanks for catching the `:` separator. 

AWS accepts character variations after `role`, so the second detail should be okay.

**Comment by bennettbuchanan** [comment]
@ironman-machine
SCALITY_INTEGRATION_BRANCH=compat/update-replication-role
try

**Comment by bennettbuchanan** [comment]
E2E passed with dependent Integration branch.

File: lib/api/apiUtils/bucket/models/ReplicationConfiguration.js [hunk 2]
```diff
@@ -114,24 +114,40 @@ class ReplicationConfiguration {
...
+        return isValidRoleARN;
+    }
+
+    /**
      * Check that the `Role` property of the configuration is valid
      * @return {undefined}
      */
     _parseRole() {
-        const Role = this._parsedXML.ReplicationConfiguration.Role;
+        const Role = this._parsedXML.ReplicationConfiguration.Role &&
```
Comment (jonathan-gramain) [inline lib/api/apiUtils/bucket/models/ReplicationConfiguration.js]: Maybe create another temp variable first for clarity

Comment (bennettbuchanan) [inline lib/api/apiUtils/bucket/models/ReplicationConfiguration.js]: Agree. I've split this into two variables for clarity and added a test for a Role value of `''`. In that case it should give an `InvalidArgument` error, not `MalformedXML`.

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine
SCALITY_INTEGRATION_BRANCH=compat/update-replication-role
try

**Comment by bennettbuchanan** [comment]
@ironman-machine
SCALITY_INTEGRATION_BRANCH=compat/update-replication-role
try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
E2E passed.

**Pull request merged**